crates/api_common/src/utils/site_utils.rs

```diff
@@ -0,0 +1,123 @@
+/// Helper functions for manipulating parts of a site.
+use lemmy_utils::error::LemmyError;
+use regex::{Regex, RegexBuilder};
+
+const SITE_NAME_MAX_LENGTH: usize = 20;
+const SITE_DESCRIPTION_MAX_LENGTH: usize = 150;
+
+/// Attempts to build the regex and checks it for common errors before inserting into the DB.
+pub fn build_and_check_regex(regex_str: &Option<&str>) -> Result<Option<Regex>, LemmyError> {
+  regex_str.map_or_else(
+    || Ok(None::<Regex>),
+    |slurs| {
+      RegexBuilder::new(slurs)
+        .case_insensitive(true)
+        .build()
+        .map_err(|e| LemmyError::from_error_message(e, "invalid_regex"))
```

Comment:
I saw that these values need to be added to the translation files, but I wasn't quite sure what to do based on the README under `/translations`. Do I need to just add the new keys to the `en.json` template and the nice folks at weblate do the rest?

Reply:
Yep that's correct. https://github.com/lemmynet/lemmy-translations